bridge_ui/src/utils/getSignedVAAWithRetry.ts

```diff
@@ -1,34 +1,42 @@
-import { ChainId, getSignedVAA } from "@certusone/wormhole-sdk";
+import {
+  ChainId,
+  ChainName,
+  getGovernorIsVAAEnqueued,
+  getSignedVAA,
+} from "@certusone/wormhole-sdk";
 import { WORMHOLE_RPC_HOSTS } from "./consts";
 
-export let CURRENT_WORMHOLE_RPC_HOST = -1;
-
-export const getNextRpcHost = () =>
-  ++CURRENT_WORMHOLE_RPC_HOST % WORMHOLE_RPC_HOSTS.length;
+export interface GetSignedVAAWithRetryResult {
+  vaaBytes: Uint8Array | undefined;
+  isPending: boolean;
+}
 
-export async function getSignedVAAWithRetry(
-  emitterChain: ChainId,
+export const getSignedVAAWithRetry = async (
+  emitterChain: ChainId | ChainName,
   emitterAddress: string,
   sequence: string,
   retryAttempts?: number
-) {
-  let result;
+): Promise<GetSignedVAAWithRetryResult> => {
+  let currentWormholeRpcHost = -1;
+  const getNextRpcHost = () =>
+    ++currentWormholeRpcHost % WORMHOLE_RPC_HOSTS.length;
   let attempts = 0;
-  while (!result) {
+  while (true) {
     attempts++;
     await new Promise((resolve) => setTimeout(resolve, 1000));
-    try {
-      result = await getSignedVAA(
-        WORMHOLE_RPC_HOSTS[getNextRpcHost()],
-        emitterChain,
-        emitterAddress,
-        sequence
-      );
-    } catch (e) {
-      if (retryAttempts !== undefined && attempts > retryAttempts) {
-        throw e;
-      }
+    const rpcHost = WORMHOLE_RPC_HOSTS[getNextRpcHost()];
+    const results = await Promise.allSettled([
```

Comment:
Just out of curiosity, why is this an array?

Reply:
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise/allSettled#parameters